xblockutils/studio_editable.py

```diff
@@ -0,0 +1,295 @@
+            'help': field.help,
+            'allow_reset': field.runtime_options.get('resettable_editor', True),
+            'list_values': None,  # Only available for List fields
+            'has_list_values': False,  # True if list_values_provider exists, even if it returned no available options
+        }
+        for type_class, type_name in supported_field_types:
+            if isinstance(field, type_class):
+                info['type'] = type_name
+                # If String fields are declared like String(..., multiline_editor=True), then call them "text" type:
+                editor_type = field.runtime_options.get('multiline_editor')
+                if type_class is String and editor_type:
+                    if editor_type == "html":
+                        info['type'] = 'html'
+                    else:
+                        info['type'] = 'text'
+                if type_class is List and field.runtime_options.get('list_style') == "set":
+                    # List represents unordered, unique items, optionally drawn from list_values_provider()
+                    info['type'] = 'set'
+                elif type_class is List:
+                    info['type'] = "generic"  # disable other types of list for now until properly implemented
+                break
+        if "type" not in info:
+            raise NotImplementedError("StudioEditableXBlockMixin currently only supports fields derived from JSONField")
+        if info["type"] in ("list", "set"):
+            info["value"] = [json.dumps(val) for val in info["value"]]
+            if info["default"] is None:
+                info["default"] = []
+            info["default"] = json.dumps(info["default"])
+        if info["type"] == "generic":
+            # Convert value to JSON string if we're treating this field generically:
+            info["value"] = json.dumps(info["value"])
+            info["default"] = json.dumps(info["default"])
+        if 'values_provider' in field.runtime_options:
+            values = field.runtime_options["values_provider"](self)
+        else:
+            values = field.values
+        if values and not isinstance(field, Boolean):
+            info['has_values'] = True
+            # This field has only a limited number of pre-defined options.
+            # Protip: when defining the field, values= can be a callable.
+            if isinstance(field.values, dict) and isinstance(field, (Float, Integer)):
+                # e.g. {"min": 0 , "max": 10, "step": .1}
+                info["min"] = field.values["min"]
+                info["max"] = field.values["max"]
+                info["step"] = field.values["step"]
+            else:
+                # e.g. [1, 2, 3] or [ {"display_name": "Always", "value": "always"}, {...}, ... ]
+                if not isinstance(values[0], dict) or "display_name" not in values[0]:
+                    values = [{"display_name": val, "value": val} for val in values]
```

Comment:
This line is supposed to convert `[1, 2, 3]` to `[{"display_name": "1", "value": 1}, {"display_name": 2, "value": 2}, {"display_name": "3", "value": 3}]`

It does work, though it looks like I forgot to force "display_name" to string type (I will fix but it wasn't an issue since the template does that anyways).